### DISCUSSION TITLE
动态的class自动推断识别问题，大家有没有遇到过，代码地址如下

### DISCUSSION THREAD

**Discussion by Miofly** [windicss/windicss#512] [General]
环境为 vue3 + vite 
"windicss": "^3.1.9",
"vite-plugin-windicss": "^1.4.11",

[动态class不识别](https://sfc.vuejs.org/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuXHQ8ZGl2IGNsYXNzPVwiZmxleCBqdXN0aWZ5LWFyb3VuZCBhbGlnbi1taWRkbGUgZmxleC13cmFwIHctWzEwMCVdXCI+XG5cdFx0PGRpdiA6Y2xhc3M9XCJbYHctWyR7d2lkdGhBdXRvfV1gXVwiIHYtZm9yPVwiKGl0ZW0sIGluZGV4KSBpbiByb3VuZGVkQXJyc1wiIDprZXk9XCJpbmRleFwiPlxuXHRcdFx0PGRpdiA6Y2xhc3M9XCJbYmFzZUNsYXNzLCBpdGVtXVwiPnt7IGl0ZW0gfX08L2Rpdj5cblx0XHQ8L2Rpdj5cblx0XHQ8ZGl2IDpjbGFzcz1cIltgdy1bJHsoMTAwIC8gcm91bmRlZEFycnMubGVuZ3RoKS50b0ZpeGVkKDIpfSVdYF1cIiB2LWZvcj1cIihpdGVtLCBpbmRleCkgaW4gcm91bmRlZEFycnNcIiA6a2V5PVwiaW5kZXhcIj5cblx0XHRcdDxkaXYgOmNsYXNzPVwiW2Jhc2VDbGFzcywgaXRlbV1cIj57eyBpdGVtIH19PC9kaXY+XG5cdFx0PC9kaXY+XG5cdDwvZGl2PlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cblx0aW1wb3J0IHsgY29tcHV0ZWQsIHJlZiB9IGZyb20gJ3Z1ZSc7XG5cblx0Y29uc3QgYmFzZUNsYXNzID0gcmVmKCd3LTgwcHggaC04MHB4IGJnLXJlZC01MDAgbS1hdXRvIHRleHQtd2hpdGUnKTtcblxuXHRjb25zdCByb3VuZGVkQXJycyA9IFsncm91bmRlZC1bNTAlXScsICdyb3VuZGVkLTEwcHgnLCAncm91bmRlZC1ub25lJ107XG5cblx0Y29uc3Qgd2lkdGhBdXRvID0gY29tcHV0ZWQoKCkgPT4ge1xuXHRcdHJldHVybiAoMTAwIC8gcm91bmRlZEFycnMubGVuZ3RoKS50b0ZpeGVkKDIpICsgJyUnO1xuXHR9KTtcbjwvc2NyaXB0PlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly91bnBrZy5jb20vQHZ1ZS9ydW50aW1lLWRvbUAzLjIuMTYvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiXG4gIH1cbn0iLCJDb21wLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgeyByZWYsIG9uTW91bnRlZCB9IGZyb20gJ3Z1ZSdcbmNvbnN0IGEgPSByZWYoMClcblxub25Nb3VudGVkKCgpID0+IHtcbiAgc2V0SW50ZXJ2YWwoKCkgPT4ge1xuICAgIGEudmFsdWUgKz0gMVxuICB9LCAyMDAwKVxufSlcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxkaXZcbiAgICBzdHlsZT1cIlxuICAgICAgYmFja2dyb3VuZDogIzE4MTkxYztcbiAgICAgIHBvc2l0aW9uOiByZWxhdGl2ZTtcbiAgICAgIG92ZXJmbG93OiBoaWRkZW47XG4gICAgICBjb2xvcjogcmVkO1xuICAgICAgYmFja2dyb3VuZC1pbWFnZTogdXJsKCdodHRwOi8vYWJjLmJjZC5jb20vZWZnLmpwZWcnKTtcbiAgICBcIlxuICA+XG4gICAge3thfX1cbiAgPC9kaXY+XG48L3RlbXBsYXRlPiJ9)

[这个颜色也存在同样的问题](https://sfc.vuejs.org/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuXHQ8ZGl2IDpjbGFzcz1cIlsnYmctcmVkLTIwMCcsICd3LTUwcHgnLCAnaC01MHB4JywgJ20tYXV0byddXCI+MTE8L2Rpdj5cblx0PGRpdiB2LWZvcj1cIihpdGVtLCBpbmRleCkgaW4gY29sb3JOdW1zXCIgOmtleT1cImluZGV4XCI+XG5cdFx0PGRpdlxuXHRcdFx0di1mb3I9XCIoc3ViSXRlbSwgc3ViSW5kZXgpIGluIGNsYXNzQXJyc1wiXG5cdFx0XHQ6a2V5PVwic3ViSW5kZXhcIlxuXHRcdFx0OmNsYXNzPVwiWydiZy0nICsgc3ViSXRlbSArICctJyArIGl0ZW0sICd3LTE1MHB4JywgJ2gtMTUwcHgnLCAnbS1hdXRvJ11cIlxuXHRcdD5cblx0XHRcdGJnLXt7IHN1Ykl0ZW0gfX0te3sgaXRlbSB9fVxuXHRcdDwvZGl2PlxuXHQ8L2Rpdj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5cdGltcG9ydCB7IHJlYWN0aXZlIH0gZnJvbSAndnVlJztcblxuXHRjb25zdCBjb2xvck51bXMgPSBbNTAsIDEwMCwgMjAwLCAzMDAsIDQwMCwgNTAwLCA2MDAsIDcwMCwgODAwLCA5MDBdO1xuXHRjb25zdCBjbGFzc0FycnMgPSByZWFjdGl2ZShbJ3BpbmsnLCAncmVkJ10pO1xuPC9zY3JpcHQ+XG5cbjxzdHlsZSBzY29wZWQ+PC9zdHlsZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vdW5wa2cuY29tL0B2dWUvcnVudGltZS1kb21AMy4yLjE2L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIlxuICB9XG59IiwiQ29tcC52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IHsgcmVmLCBvbk1vdW50ZWQgfSBmcm9tICd2dWUnXG5jb25zdCBhID0gcmVmKDApXG5cbm9uTW91bnRlZCgoKSA9PiB7XG4gIHNldEludGVydmFsKCgpID0+IHtcbiAgICBhLnZhbHVlICs9IDFcbiAgfSwgMjAwMClcbn0pXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZGl2XG4gICAgc3R5bGU9XCJcbiAgICAgIGJhY2tncm91bmQ6ICMxODE5MWM7XG4gICAgICBwb3NpdGlvbjogcmVsYXRpdmU7XG4gICAgICBvdmVyZmxvdzogaGlkZGVuO1xuICAgICAgY29sb3I6IHJlZDtcbiAgICAgIGJhY2tncm91bmQtaW1hZ2U6IHVybCgnaHR0cDovL2FiYy5iY2QuY29tL2VmZy5qcGVnJyk7XG4gICAgXCJcbiAgPlxuICAgIHt7YX19XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT4ifQ==)

**Comment by 16bitcode** [comment]
我也遇到了，最后怎么解决的

**Reply by Miofly** [reply]
这个功能本身就是不支持的，除非你把这些类名一开始就注入到白名单当中

**Reply by 16bitcode** [reply]
不对啊，我记得这些功能是应该有的，我需要动态设置定位的 left、top

**Reply by Miofly** [reply]
> 不对啊，我记得这些功能是应该有的，我需要动态设置定位的 left、top

仔细看文档。它生成class机制就不是动态，
